main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -97,4 +98,37 @@ object TypeReduction {
     case Type.Alias(cst, args, t, _) => simplify(t, renv0, loc)
   }
 
+  def simplifyJava(tpe: Type, loc: SourceLocation)(implicit flix: Flix): Result[(Type, Boolean), TypeError] =
+    tpe.typeConstructor match {
+      case Some(TypeConstructor.MethodReturnType(m, n)) =>
+        val targs = tpe.typeArguments
+        resolveMethodReturnType(targs.head, m, targs.tail, loc) match {
+          case ResolutionResult.Resolve(t) => Result.Ok((t, true))
+          case ResolutionResult.NotFound() => ???
```

Comment:
Let us add a new TypeError.MethodNotFound. Have a look in TypeError and take inspirations from the existing errors.